---
title: "Algatr"
format: pdf
editor: visual
---

-   algatr help website: <https://thewanglab.github.io/algatr/>

-   algatr github: <https://github.com/TheWangLab/algatr_workshop>

```{r}

library(installr) 
install.packages.zip("https://cran.csiro.au/bin/macosx/big-sur-x86_64/contrib/4.5/wingen_2.2.0.tgz")


cp -R -i -n /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/ /Library/Frameworks/R.framework/Resources/library
```

\^ for installing packages using zip file from cran Installing it just using base r `install.packages` (wasn't working for some reason, it was returning a 404 error page)

## Algatr Tutorial

```{r}
library(algatr)
# alazygatr_packages()
    # ^updates packages

#Warning message:
#In i.p(...) :
#  installation of package ‘/var/folders/5g/bpbtlbnn6n9_ptfrqwyy_wkh0000gn/T//RtmpsX4vU0/filea4c36bd44d75/RStoolbox_1.0.2.1.tar.gz’ had non-zero exit status

load_algatr_example()
plot(CA_env, col = turbo(100), axes = FALSE)

env <- scaleRGB(CA_env)
plotRGB(env, r = 1, g = 2, b = 3)

# Add sampling localities on top of this
points(liz_coords, pch = 19)

```

### Files needed:

-   `do_everything_for_me(liz_vcf, liz_coords, CA_env)`

-   vcf file, coordinate file (long=x, lat=y), and env file

env file

-   rasterstack object (<https://rspatial.github.io/raster/reference/stack.html>)

-   Example env file

    -   These were generated by performing a raster PCA on 19 bioclimatic variables (obtained from the World Clim database) and retaining the top 3 PCs

    -   We can combine all three PCs into a single map by scaling each of the rasters such that they each correspond to either R, G, or B using the [`scaleRGB()`](https://thewanglab.github.io/algatr/reference/scaleRGB.html) function and subsequently map using the `plotRGB()` function.

        ```{r}
        env <- scaleRGB(CA_env)
        plotRGB(env, r = 1, g = 2, b = 3)
        ```

### Getting Raster data into R

Tutorials:

-   <https://wec.wur.nl/r/spatial/raster-data.html>

-   <https://ouicodedata.com/posts/worldclim-with-r/>

```{r}
library(raster)
library(sp)
library(geodata)
library(algatr)

raster_path = "/Users/caprinapugliese/Documents/School/Uconn/2024-26_Grad_School/Dagilis-lab/WNS-project/data/climate_data/wc2.1_data"

bio <- worldclim_global(path=raster_path, var="bio", res=2.5)

# information on the coordinate system
crs(CAN_bio, proj = TRUE, describe = TRUE)
crs(USA_bio, proj = TRUE, describe = TRUE)

plot(bio[[12]])
plot(CAN_bio[[12]])

# crop Raster* with Spatial* object
NoA <- as(extent(-135, -55, 25, 60), 'SpatialPolygons')
crs(NoA) <- crs(bio)
NoA_bio <- crop(bio, NoA)
plot(NoA_bio[[12]])

## https://stackoverflow.com/questions/75739085/running-a-pca-on-a-rasterstack-in-r
## pca code
pca <- prcomp(NoA_bio)
x <- predict(NoA_bio, pca)
plot(x)

NoA_pca_map <- plotRGB(scaleRGB(x),r=1,g=2,b=3)
```

```{r}
# example using multiple countries
w <-  worldclim_country(c("BEL", "NL", "Luxembourg"), var="tavg")
m <- merge(w)
plot(bio_m, 1)

# plot the 12 bio element
plot(bio[[12]])


library(ggplot2)
ggplot(bio) +
  geom_sf(fill = "#69b3a2", color = "white") +
  theme_void()

library(leaflet)
map <- addTiles(leaflet())
# save the widget in a html file if needed.
# library(htmlwidgets)
# saveWidget(m, file=paste0( getwd(), "/HtmlWidget/backgroundMapBasic.html"))

map 
```

-   data

    -   x –\> long min: -121.5, max: -62

    -   y –\> lat min: 34.5, max: 48.5

## Using TESS

-   recommend using `ld_prune()` algatr function first

    -   "ld_prune prunes SNPs based on linkage disequilibrium using SNPRelate and SeqArray packages"

        ```{r}
        tess_packages()



        ld_prune(vcf=vcf,out_name="ld_prun_vcf",out_format="vcf",ld.threshold = 0.6,method = c("corr"),save_output = TRUE)
        # vcf is biallelic
        # this command was not working
            # first couldn't find library "here" and had to install that
            # then got this error:
                # Error in `SNPRelate::snpgdsVCF2GDS()`:
                # is.character(vcf.fn) & is.vector(vcf.fn) is not TRUE

        gt <- extract.gt(vcf)
        ```

        -   ended up just ld pruning with plink and outputting a vcf file and transferring that over instead

            ```{r}
            # north america no clones, filtered vcf
            File attributes:
              meta lines: 120
              header_line: 121
              variant count: 506864
              column count: 60

            # north america no clones, filtered, ld pruned vcf
            File attributes:
              meta lines: 102
              header_line: 103
              variant count: 1176
              column count: 60

            # only the regular vcf has NA values, the prunned one does not
            > any(is.na(pruned_dosage))
            [1] FALSE
            > any(is.na(dosage))
            [1] TRUE
            > any(is.na(simple_dos))
            [1] FALSE
            ```

linkage disequalibrium:

"For other analyses - such as genotype-environment association methods like RDA and LFMM - SNPs in LD are sometimes removed prior to running such analyses, but we do not recommend this because SNPs most strongly associated with environmental variables may have been pruned out. However, failing to account for linkage may result in overinflated conclusions about environmental associations. A solution to this is to perform LD-pruning after having run GEA methods; one approach to do so is to retain SNPs with the strongest association (i.e., lowest p-value) and removing all others that are in linkage with that particular SNP (see the approach described in Capblancq & Forester 2021 for more information)" -algatr tutorial