---
title: "Downloading and Processing Sequences"
format: pdf
editor: visual
---

## Using SRA toolkit to download files

-   create data working directory

-   preset download to working directory

    `vdb-config --prefetch-to-cwd`

-   download the file

    `prefetch [SRA-id-number]`

## FastQC

-   use fasterq-dump to extract the file

    `fasterq-dump [SRA-id-number] --split-files`

    `fastqc --noextract [fastq-split-file-1] [fastq-split-file-2] [...]`

## Trimming seq with bbduk

``` bash
bbduk.sh in1=SRR11812088_1.fastq in2=SRR11812088_2.fastq out1=SRR11812088_1.trimmed.fastq out2=SRR11812088_2.trimmed.fastq t=10 qtrim=rl maq=10 ftl=10 ftr=-139 trimq=15 tpe tbo
```

## Mapping seq to genome

using bwa mem

```         
bwa mem -t 8 pdestructans.fasta SRR11812088_1.trimmed.fastq SRR11812088_2.trimmed.fastq | samtools sort -@8 -o SRR11812088-aligned.bam
```

## Editing bam file

```         
java -jar ~/software/picard.jar AddOrReplaceReadGroups \
      I=SRR11812088-aligned.bam \
      O=SRR11812088.named.bam \
      RGID=1 \
      RGLB=lib1 \
      RGPL=ILLUMINA \
      RGPU=unit1 \
      RGSM=test
```

```         
java -jar ~/software/picard.jar MarkDuplicates \
I=SRR11812088.named.bam \
O=SRR11812088.dedup.bam \
M=dedup.txt
```

```         
samtools sort SRR11812088.dedup.bam -o SRR11812088.sorted.bam

samtools index SRR11812088.sorted.bam
```

## Calling genotypes

using bcftools

```{bash}
bcftools mpileup alignments/SRR11812088.sorted.bam --fasta-ref pd_ref/pdestructans.fasta | bcftools call -m --ploidy 1 -Oz -o test.g.vcf.gz
```

using GATK pipeline

```{bash}
gatk HaplotypeCaller -R pd_ref/pdestructans.fasta -I alignments/SRR11812088.sorted.bam -O test2.gvcf.gz -ERC GVCF -ploidy 1

gatk GenotypeGVCFs -R pd_ref/pdestructans.fasta \
  -V test2.gvcf.gz \
  -O test2.vcf.gz
```

output:

-   Either a VCF or GVCF file with raw, unfiltered SNP and indel calls. Regular VCFs must be filtered either by variant recalibration (Best Practice) or hard-filtering before use in downstream analyses. If using the GVCF workflow, the output is a GVCF file that must first be run through GenotypeGVCFs and then filtering before further analysis.