---
title: "10-09-24_terminal-notes"
format: pdf
editor: visual
---

### 10-9

-   successfully logged onto HPC storrs

-   started an interactive session

    -   session quit unexpectedly due to "broken pipe"

    -   logged back in and cancelled the session then created a new session

-   created a new data directory, new temporary data folder

    -   trying to download sequence data

    -   seeing if I can use a python script shortcut to download the data

    -   had to load in python

        -   uses pip to install package, but the pip command is not working

        -   pip is installed, so idk why its not working

-   session timed out bc I had it set to only last one hour, so I can try to figure this out another day

    -   <https://www.geeksforgeeks.org/how-to-fix-pip-command-not-found-in-linux/>

### 10-10

-   made a shortcut to pip in my directory

    -   '\_\_main\_' not found error

    -   gave up

    -   SRA toolkit might be able to download it, i may be an idiot

-   storrs HPC doesnt even look like they have SRA toolkit installed

    -   may have to use farmington cluster for sequence stuff or download sratoolkit to directory

### 10/11

-   logged onto xanadu cluster

-   able to load in sra toolkit

-   created dir: data/sra_seq

-   set prefetch to new directory

-   prefetched SRX8363356

-   extracted file with fastq

-   tried to analyze using fastqc but run was killed

    -   generated a fastqc zip file but not a viewable html file

    -   looks like I may have run out of memory

    -   next time try initiating an interactive session with 2G of memory

### 10/15

-   was able to successfully generate html

-   unable to open html on cluster

    -   <https://stackoverflow.com/questions/21124869/how-to-view-html-file-in-remote-unix-server>

    -   tried to open it with a local host --\> did not work

    -   tried trouble shooting --\> still did not work

-   now trying to mirror directory from cluster to current computer (maybe look into file transfering???)

    -   downloaded macfuse and SSHFS software

    -   <https://github.com/osxfuse/osxfuse/wiki/SSHFS>

### 10/16

-   tried to use filezilla to transfer files

    -   filezilla was unable to connect to xanadu transfer node

-   trying to figure out how the SSHFS software works

    -   OMG IT WORKS!!

``` bash
sshfs cpugliese@transfer.cam.uchc.edu:tmp/data/sra-seq/fastq /users/caprinapugliese/Documents/School/Uconn/2024-26_Grad_School/Dagilis-lab/cluster-mirror
```

tmp/data/sra-seq/fastq

/users/caprinapugliese/Documents/School/Uconn/2024-26_Grad_School/Dagilis-lab/cluster-mirror

\--

### 10/24

-   edited .bashrc on both laptop and xanadu

-   loaded in more modules (and added to .bashrc) to help with mapping the sequences to the reference genome

-   ran bwa-mem code

    `bwa mem -t 8 Pd-ref-genome SRR11812088_1.trimmed.fastq SRR11812088_2.trimmed.fastq | samtools sort -@8 -o SRR11812088.bam`

-   just finished adding picard to the modules list

### 10/30

-   had trouble getting the picard command to work

    -   had to use java command and tell the computer where picard was

-   successfully edited bam file

-   tried to map to genome, but ran into issues

### 11/6

-   found out during lab meeting that the genome I downloaded was a zip file

    -   might have to go back and do bwa step

    -   also might have to index the genome before I try to map it (samtools FAIDX or samtools index)

### 11/7

-   trying to figure out how to index the genome (it's not working)

-   error saying index files missing

-   don't know if it's an issue with the download or not, but I cannot troubleshoot anymore today because it feels like I am slamming my brain against a brick wall rn

### 11/13

/labs/Dagilis/Projects/WNS/ref

### 11/19

-   made symlinks to get to lab directory and to Pd genome ref folder

-   moved the extra bam files I created last time to .trash/

-   creating seq dictionary:

    -   `java -jar ~/software/picard.jar CreateSequenceDictionary R=pdestructans.fasta O=pdestructans.dict`

    -   the dictionary I created appears to be larger than the one in the lab reference folder??? (12,677 vs 9,772)

-   trying to remake .bam file with bwa mem

    -   /home/FCAM/cpugliese/tmp/data/sra-seq/downloads/brickwall

```{bash}
bwa mem -t 8 pdestructans.fasta SRR11812088_1.trimmed.fastq SRR11812088_2.trimmed.fastq | samtools sort -@8 -o SRR11812088-aligned.bam
```

-   finally got it to work

-   need to go back and re-do all the .bam editing (everything in alignments prob needs to go into .trash/

-   need to move brickwall/ out of downloads and rename to reference genome or something

-   probably need to clean up a bit

### 11/20

-   cleaned up and reorganized things

-   edited bam file

-   still ran into same "failed to open file" error

    -   looks like I was able to fix it by telling the code the exact location of the files

        ```{bash}
        bcftools mpileup alignments/SRR11812088.sorted.bam --fasta-ref pd_ref/pdestructans.fasta | bcftools call -m --ploidy 1 -Oz -o test.g.vcf.gz
        ```

-   last thing left to do is the gatk pipeline

-   gatk isnt a downloaded module --\> was able to download it from github

    -   am not able to get the command to work yet though
